File: java/type1/calculate_average_comments.java
class CalculateAverageComments {
    // Computes mean value over an array
    static double calculateAverage(double[] numbers) {
        // Sum all elements
        double total = 0;
        for (double x : numbers) total += x;
        // Count elements
        int count = numbers.length;
        // Return average
        return total / count;
    }
}

File: test/javascript/type1/CalculateAverageComments.js
// Computes mean value
function calculateAverage1b(numbers) {
    const total = numbers.reduce((a, b) => a + b, 0); // Sum all elements
    const count = numbers.length; // Count elements
    return total / count; // Return average
}

module.exports = { calculateAverage1b };
